src/harness/testcases/grant_testcase.py

```diff
@@ -624,3 +624,245 @@ def test_WINNF_FT_S_GRA_28(self):
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+    # Create and trigger the ESC Zone
+    exclusion_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'exclusion_zone_0.json')))
+    zone_response = self._sas_admin.InjectEscZone(exclusion_zone)
+    trigger_esc_zone_request = {'zoneId': zone_response['zoneId'],
+                                    'frequencyRange': {
+                                     'lowFrequency': 3620000000.0,
+                                     'highFrequency': 3630000000.0}}
+    trigger_id = self._sas_admin.TriggerEscZone(trigger_esc_zone_request)
+    # Request grant
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    # Check grant response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertTrue(response['grantId'])
+    self.assertEqual(response['channelType'], 'GAA')
+    self.assertEqual(response['response']['responseCode'], 0)
+
+  @winnforum_testcase
+  def test_WINFF_FT_S_GRA_3(self):
+    """Successful CBSD grant request.
+    Federal Incumbent is present in the GAA frequency range requested by
+    the CBSD which is inside the protection zone of Federal Incumbent.
+    """
+
+    # Register the device inside the protection zone
+    device_inside_esc_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    self._sas_admin.InjectFccId({'fccId': device_inside_esc_zone['fccId']})
+    request = {'registrationRequest': [device_inside_esc_zone]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+    # Create and trigger the ESC Zone
+    exclusion_zone = json.load(
+        open(os.path.join('testcases', 'testdata', 'exclusion_zone_0.json')))
+    zone_response = self._sas_admin.InjectEscZone(exclusion_zone)
+    trigger_esc_zone = {'zoneId': zone_response['zoneId'],
+                                    'frequencyRange': {
+                                     'lowFrequency': 3620000000.0,
+                                     'highFrequency': 3630000000.0}}
+    trigger_id = self._sas_admin.TriggerEscZone(trigger_esc_zone)
+    # Request grant
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+    response = self._sas.Grant(request)['grantResponse'][0]
+    # Check grant response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    if(response['response']['responseCode'] == 400) :
```

Comment:
Parentheses can be removed in the if condition. Also, can the condition be changed, instead of checking for 400 can we check for 0 because Test Spec mentions the check for ResponseCode 0 and will be easy to verify with Spec.